### PR TITLE
Fixing wrong sitemap product url #4788

### PR DESCRIPTION
**Pull request by DenisSaltanahmedov**
### Description
Fixed issue #4788
https://github.com/magento/magento2/issues/4788

### Fixed Issues (if relevant)
Fixed wrong sitemap product URL

### Manual testing scenarios
1. "Use Categories Path for Product URLs" set to NO in configuration -> catalog -> catalog -> Search  Engine Optimization
2. Create Categories with products
3. Generate sitemap

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=14338) <br/>All committers have signed the CLA.

File: app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php [hunk 1]
```diff
@@ -88,6 +91,12 @@ class Product extends \Magento\Framework\Model\ResourceModel\Db\AbstractDb
      * @var \Magento\Catalog\Helper\Image
      */
     private $catalogImageHelper;
+    /**
+     * Scope Config
+     *
+     * @var ConfigInterface
+     */
+    protected $scopeConfig;
```
Comment (sidolov) [inline app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php]: It's would be better use private property here

File: app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php [hunk 2]
```diff
@@ -114,6 +124,7 @@ public function __construct(
         \Magento\Catalog\Model\ResourceModel\Product\Gallery $mediaGalleryResourceModel,
         \Magento\Catalog\Model\Product\Gallery\ReadHandler $mediaGalleryReadHandler,
         \Magento\Catalog\Model\Product\Media\Config $mediaConfig,
+        ConfigInterface $scopeConfig,
```
Comment (sidolov) [inline app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php]: Please, inject new dependencies according to our [backward compatibility policy](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)

File: app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php [hunk 3]
```diff
@@ -450,4 +467,20 @@ private function getProductImageUrl($image)
...
     }
+
+    /**
+     * Return Use Categories Path for Product URLs config value
+     *
+     * @param $storeId
+     *
+     * @return bool
+     */
+    protected function getCategoryProductURLsConfig($storeId)
```
Comment (sidolov) [inline app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php]: This method is not a part of api, so, make it private, please

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by magento-engcom-team** [comment]
@DenisSaltanahmedov thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php [hunk 4]
```diff
@@ -114,6 +124,7 @@ public function __construct(
         \Magento\Catalog\Model\ResourceModel\Product\Gallery $mediaGalleryResourceModel,
         \Magento\Catalog\Model\Product\Gallery\ReadHandler $mediaGalleryReadHandler,
         \Magento\Catalog\Model\Product\Media\Config $mediaConfig,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
```
Comment (sidolov) [inline app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php]: Please, inject new dependencies according to our [Backward compatible development policy](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter)

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @DenisSaltanahmedov , please, fix issues from the review and take a look to failed static test

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-1073](https://jira.corp.magento.com/browse/ENGCOM-1073) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @DenisSaltanahmedov. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.5 release.

**Pull request merged**